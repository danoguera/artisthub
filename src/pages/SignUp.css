.signupFrame {
  display: flex;
  justify-content: center;
  margin-top: 5px;
  text-align: center;
  color: #FFFFFF;
  /* line-height: 20px; */
  /* top: 50%;
  left: 50%; */
  /* transform: translate(-50%,-50%);
  height: 500px; */
}

.boxSignup{
  background-color: #E8E8E8;
  width: 450px;
  height: 780px;
  margin-top: 20px;
  background: rgba(0,0,0,0.4);
  padding: 10px;
  margin-bottom: 10px;
  border-radius: 8px;
}

.boxSignup h1{
  font-size: 30px;
}

.boxSignup h2{
  font-size: 20px;
}

.boxSignup select{
  font-size: 14px;
  display: block;
  background: rgba(240,240,240,0.05);
  color: #FFFFFF;
  padding: 10px 35px;
  width: 89%;
  max-width: 100%;
  margin: 20px;
  border-radius: 0.5em;
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  line-height: 10px;
  border: 1px solid rgb(199, 190, 190);
  background-image: url('../assets/images/down-arrow.png');
  background-repeat: no-repeat, repeat;
  background-position: right .7em top 50%;
  background-size: .90em;
  font-size: 18px;
}

.boxSignup select option {
  color: black;
}

input[type="file"]{
  display: none;
}

.boxSignup label{
  display: flex;
  justify-content: center;
  align-items: center;
  background: rgba(240,240,240,0.05);
  border: 1px solid rgb(199, 190, 190);
  border-radius: 6px;
  line-height: 2px;
  /* margin-top: 10px; */
  margin-left: 17px;
  width: 89%;
}

.boxSignup label p{
  margin-left: 6px;
  font-size: 14px;
}

::placeholder{
  color: #FFFFFF;
  font-size: 18px;
}

.boxSignup p {
  display: flex;
  justify-content: left;
  /* text-align: right; */
  /* background: rgba(240,240,240, 0); */
  border: transparent;
  border-radius: 6px;
  padding: 25px 40px;
  margin: 0 20px;
  width: 86.5%;
  padding: 30px;
  /* font-weight: lighter; */
  font-size: 18px;
  color: #ffffff;
  line-height: 30px;
}

.date{
  color: #FFFFFF;
  font-size: 18px;
  background: rgba(240,240,240, 0);
  border: transparent;
  border-radius: 6px;
  padding-left: 10px;
}


.inputBoxSignup{
  display: block;
  box-sizing: border-box;
  margin: 0 10px;
  padding: 30px;
  background: rgba(240,240,240,0.05);
  width: 89%;
  height: 32px;
  border: 1px solid rgb(199, 190, 190);
  outline: none;
  font-size: 13px;
  transition: 0.2s;
  color: #FFFFFF;
  border-radius: 5px;
  line-height: 40px;
}

.textBox{
  display: block;
  box-sizing: border-box;
  margin: auto;
  padding: 10px;
  background: transparent;
  width: 90%;
  height: 92px;
  border: 1px solid #FFF;
  outline: none;
  font-size: 15px;
  transition: 0.2s;
  color: #FFF;
  /* text-shadow: 2px 2px 8px #fff; */
  font-weight: 400;
  border-radius: 5px;
}

.submit-btn{
  width: 120px;
  height: 32px;
  background: #F44336;
  border: none;
  border-radius: 8px;
  color: #FFF;
  cursor: pointer;
  transition: 0.2s;
  font-size: 20px;
}

@media only screen and (min-width: 1000px){

  .signup {
    height: 400px;
    margin-top: 10px;
  }

  .boxSignup{
    background-color: #E8E8E8;
    width: 1000px;
    height: 920px;
    /* margin: auto; */
    background: rgba(0,0,0,0.4);
    padding: 10px;
    border-radius: 8px;
    margin-bottom: 10px;
  }
  
  .boxSignup h1{
    font-size: 30px;
  }
  
  .boxSignup h2{
    font-size: 20px;
  }
  
  .boxSignup select{
    display: block;
    background: rgba(240,240,240,0.05);
    color: #FFFFFF;
    padding: 10px 15px;
    width: 89%;
    border-radius: 6px;
    line-height: 20px;
    margin: 0 auto;
    border: 1px solid rgb(199, 190, 190);
    background-image: url('../assets/images/down-arrow.png');
    background-repeat: no-repeat, repeat;
    background-position: right .7em top 50%;
    background-size: .90em;
  
  }

  .boxSignup p{
    display: flex;
    justify-content: left;
    /* text-align: right; */
    background: rgba(240,240,240,0.05);
    border: 1px solid rgb(199, 190, 190);
    border-radius: 6px;
    padding: 5px 0;
    margin: auto;
    width: 87.5%;
    padding-left: 15px;
  }

  input[type="file"]{
    display: none;
  }
  
  .boxSignup label{
    display: flex;
    justify-content: center;
    align-items: center;
    background: rgba(126, 82, 82, 0.05);
    border: 1px solid rgb(199, 190, 190);
    border-radius: 6px;
    line-height: 2px;
    margin: auto;
    width: 89%;
  }
  
  .boxSignup label p{
    margin-left: 6px;
    font-size: 14px;
  }
  
  ::placeholder{
    color: #FFFFFF;
  }
  
  .inputBoxSignup{
    display: block;
    box-sizing: border-box;
    margin: auto;
    padding: 20px 15px;
    background: rgba(240,240,240,0.05);
    width: 89%;
    height: 32px;
    border: 1px solid rgb(199, 190, 190);
    outline: none;
    font-size: 15px;
    transition: 0.2s;
    color: #FFFFFF;
    border-radius: 5px;
    line-height: 50px;
  }
  
  .textBox{
    display: block;
    box-sizing: border-box;
    margin: auto;
    padding: 10px;
    background: transparent;
    width: 89%;
    height: 92px;
    border: 1px solid #FFF;
    outline: none;
    font-size: 15px;
    transition: 0.2s;
    color: #FFF;
    font-weight: 400;
    border-radius: 5px;
  }

  
}