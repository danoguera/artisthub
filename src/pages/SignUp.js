import React from 'react';
import axios from 'axios';
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import './SignUp.css';

class SignUp extends React.Component{
    constructor(){
        super();
        this.state = {
            name: "",
            lastname: "",
            username: "",
            email: "",
            password: "",
            verifyPassword: "",
            documentId: "",
            result: "",
            country: "",
            state: "",
            city: "",
            birthDate: new Date("January 1, 2001"),
            notEqual: false,
            typeOfUser: "",        
        } 
    } 


    handleSelect = (event) => {
        this.setState({
           [event.target.name]: event.target.value 
        })
    } 

    handleSubmit = (event) => {
        event.preventDefault();
        let url;
        if (this.state.typeOfUser==="provider"){
            url= process.env.REACT_APP_SERVER_URL+"/provider/signup";
        } else{
            url= process.env.REACT_APP_SERVER_URL+"/users/signup";
        } 
        const {name, lastname, password, state, country, city, email, username, documentId, birthDate} = this.state; 
        axios({
            url,
            method: "POST",
            headers: {  },
            data: {
                name,
                lastname,
                username,
                email,
                password,
                city,
                state,
                country,
                documentId,
                birthDate,
            }
            })
            .then(response => {
                alert("Tu usuario ha sido registrado")
                this.props.history.push("/home");  
            })
            .catch(error =>{
                alert("No es posible crear/modificar el usuario. " + error.response.data.message);
                 this.setState({ 
                    error: error.response.data.message,
                })
             })
            .finally(() => this.setState({ loading: false })); 
    
    } 

    handleInput = (event) => {
        this.setState({
           [event.target.name]: event.target.value, 
        }) 
        if (event.target.name==="verifyPassword"){
            if (event.target.value !== this.state.password) {
                this.setState({ notEqual: true });
            }else {
                this.setState({ notEqual: false });
            }
        }
    } 
    handleChangeDate = date => {
        this.setState({
          birthDate: date
        });
      };

    render(){
        if (this.state.loading){return <h1>Loading...</h1>}
      
        return (
            <main>
                <section>
                    <div class="signup">
                        <div class="boxSignup">
                            <h1>Create your profile</h1>
                            <h2>Please fill up the form</h2>
                            <form onSubmit={this.handleSubmit} >
                                <select onChange={this.handleInput} value={this.state.typeOfUser} name="typeOfUser" className="inputBoxSignup" id="typeOfUser">
                                    <option value="">Select your type of user:</option>
                                    <option value="user">User</option>
                                    <option value="provider">Provider</option>
                                </select>
                                <br/>
                                <input id="name" placeholder="Type your name" name="name" type="text" className="inputBoxSignup" value={this.state.name} onChange={this.handleInput} />
                                <br/>
                                <input id="lastname" placeholder="Type your lastname" name="lastname" type="text" className="inputBoxSignup" value={this.state.lastname} onChange={this.handleInput} />
                                <br/>
                                <input id="password" placeholder="Type your password" name="password" type="password" className="inputBoxSignup" value={this.state.password} onChange={this.handleInput} />
                                <br/>
                                <input id="verifyPassword" placeholder="Verify your password" name="verifyPassword" type="password" className="inputBoxSignup" value={this.state.verifyPassword} onChange={this.handleInput} />
                                <br/>
                                <input id="email" placeholder="Type your email" name="email" type="text" className="inputBoxSignup" value={this.state.email} onChange={this.handleInput} />
                                <br/>
                                <input id="username" placeholder="Type your username" name="username" type="text" className="inputBoxSignup" value={this.state.username} onChange={this.handleInput} />
                                <br/>
                                <input id="documentId" placeholder="Type your Document ID" name="documentId" type="text" className="inputBoxSignup" value={this.state.documentId} onChange={this.handleInput} />
                                <br />
                                <p>Birthdate: <DatePicker selected={this.state.birthDate} onChange={this.handleChangeDate} dateFormattttt="yyyy/MM/dd" className="date" /></p>
                                <br />
                                <select onChange={this.handleInput} value={this.state.country} name="country" class="countries" id="countryId">
                                    <option value="">Select Country</option>
                                    <option value="Australia">Australia </option>
                                    <option value="Colombia">Colombia</option>
                                </select>
                                <br/>
                                <select onChange={this.handleInput} value={this.state.state} name="state" class="states" id="stateId">
                                    <option value="">Select State</option>
                                    <option value="Victoria">Victoria</option>
                                    <option value="Atlantico">Atlantico</option>
                                    <option value="Bogota">Bogota</option>
                                </select>
                                <br/>
                                <select onChange={this.handleInput} value={this.state.city} name="city" class="cities" id="cityId">
                                    <option value="">Select your city:</option>
                                    <option value="Barranquilla">Barranquilla</option>
                                    <option value="Bogota">Bogota</option>
                                    <option value="Bucaramanga">Bucaramanga</option>
                                    <option value="Cali">Cali</option>
                                    <option value="Medellin">Medellin</option>
                                    <option value="Melbourne">Melbourne</option>
                                </select>
                                <br/>
                                <input type="submit" onSubmit={this.handleSubmit} className="submit-btn" value={this.state.postId ? "Update" : "Submit"} />
                                <br/>
                            </form>
                        </div>
                    </div>
                </section>
            </main>
        )

    }
} 

export default SignUp;
