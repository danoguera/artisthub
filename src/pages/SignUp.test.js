import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import moxios from 'moxios';
import SignUp from './SignUp';

describe('SignUp', () => {
  beforeEach(() => moxios.install());
  
  afterEach(() => {
    moxios.uninstall();
    cleanup();
  });

  it('should render correctly', () => {
    const { container } = render(<SignUp />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should change data as user writes in the input boxes ', (done) => {
    const { getByTestId, getByPlaceholderText } = render(
        <SignUp  />
    );

    const input = getByPlaceholderText('Type your name');
    const event = {
      target: {
        name: 'name',
        value: 'Peter',
      },
    };

    fireEvent.change(input, event); 

    expect(getByTestId('name').value).toBe("Peter");
    done();
  });

  it('should change type of User  ', (done) => {
    const { getByTestId, getByPlaceholderText } = render(
        <SignUp  />
    );

    const input = getByTestId('typeOfUser');
    const event = {
      target: {
        name: 'typeOfUser',
        value: 'provider',
      },
    };

    fireEvent.change(input, event); 

    expect(getByTestId('typeOfUser').value).toBe("provider");
    done();
  });

  it('should jump to home after user signup', (done) => {
    const history = [];
    window.alert = jest.fn();

    const { getByTestId, getByPlaceholderText } = render(
        <SignUp history={history}/>
    );

    fireEvent.click(getByTestId('submit-btn'));

    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
       status: 200,
       response: {"message":"OK"}
        }).then(() => {
       
       expect(history[0]).toMatch("/home"); 
       done();
      })
    });

  });

  it('should validate type of User at signup', (done) => {
    const history = [];
    window.alert = jest.fn();

    const { getByTestId, getByPlaceholderText } = render(
        <SignUp history={history}/>
    );

    const input = getByTestId('typeOfUser');
    const event = {
      target: {
        name: 'typeOfUser',
        value: 'provider',
      },
    };

    fireEvent.change(input, event); 


    fireEvent.click(getByTestId('submit-btn'));

    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
       status: 200,
       response: {"message":"OK"}
        }).then(() => {
       
       expect(history[0]).toMatch("/home"); 
       done();
      })
    });

  });

  it('should show an error if signup fails', (done) => {
    const history = [];
    window.alert = jest.fn();

    const { getByTestId, getByPlaceholderText } = render(
        <SignUp history={history}/>
    );

    fireEvent.click(getByTestId('submit-btn'));

    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
       status: 400,
       response: {"message":"Not Ok"}
        }).then(() => {
          expect(window.alert).toHaveBeenCalled();  //Esto es correcto?
       done();
      })
    });

  });

})