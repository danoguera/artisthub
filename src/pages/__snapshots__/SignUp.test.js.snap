// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should render correctly 1`] = `
<main>
  <section>
    <div
      class="signup"
    >
      <div
        class="boxSignup"
      >
        <h1>
          Create your profile
        </h1>
        <h2>
          Please fill up the form
        </h2>
        <form>
          <select
            class="inputBoxSignup"
            data-testid="typeOfUser"
            id="typeOfUser"
            name="typeOfUser"
          >
            <option
              value=""
            >
              Select your type of user:
            </option>
            <option
              value="user"
            >
              User
            </option>
            <option
              value="provider"
            >
              Provider
            </option>
          </select>
          <br />
          <input
            class="inputBoxSignup"
            data-testid="name"
            id="name"
            name="name"
            placeholder="Type your name"
            type="text"
            value=""
          />
          <br />
          <input
            class="inputBoxSignup"
            id="lastname"
            name="lastname"
            placeholder="Type your lastname"
            type="text"
            value=""
          />
          <br />
          <input
            class="inputBoxSignup"
            id="password"
            name="password"
            placeholder="Type your password"
            type="password"
            value=""
          />
          <br />
          <input
            class="inputBoxSignup"
            id="verifyPassword"
            name="verifyPassword"
            placeholder="Verify your password"
            type="password"
            value=""
          />
          <br />
          <input
            class="inputBoxSignup"
            id="email"
            name="email"
            placeholder="Type your email"
            type="text"
            value=""
          />
          <br />
          <input
            class="inputBoxSignup"
            id="username"
            name="username"
            placeholder="Type your username"
            type="text"
            value=""
          />
          <br />
          <input
            class="inputBoxSignup"
            id="documentId"
            name="documentId"
            placeholder="Type your Document ID"
            type="text"
            value=""
          />
          <br />
          <p>
            Birthdate: 
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class="date"
                  type="text"
                  value="01/01/2001"
                />
              </div>
            </div>
          </p>
          <br />
          <select
            class="countries"
            id="countryId"
            name="country"
          >
            <option
              value=""
            >
              Select Country
            </option>
            <option
              value="Australia"
            >
              Australia 
            </option>
            <option
              value="Colombia"
            >
              Colombia
            </option>
          </select>
          <br />
          <select
            class="states"
            id="stateId"
            name="state"
          >
            <option
              value=""
            >
              Select State
            </option>
            <option
              value="Victoria"
            >
              Victoria
            </option>
            <option
              value="Atlantico"
            >
              Atlantico
            </option>
            <option
              value="Bogota"
            >
              Bogota
            </option>
          </select>
          <br />
          <select
            class="cities"
            id="cityId"
            name="city"
          >
            <option
              value=""
            >
              Select your city:
            </option>
            <option
              value="Barranquilla"
            >
              Barranquilla
            </option>
            <option
              value="Bogota"
            >
              Bogota
            </option>
            <option
              value="Bucaramanga"
            >
              Bucaramanga
            </option>
            <option
              value="Cali"
            >
              Cali
            </option>
            <option
              value="Medellin"
            >
              Medellin
            </option>
            <option
              value="Melbourne"
            >
              Melbourne
            </option>
          </select>
          <br />
          <input
            class="submit-btn"
            data-testid="submit-btn"
            type="submit"
            value="Submit"
          />
          <br />
        </form>
      </div>
    </div>
  </section>
</main>
`;
